### PR TITLE
Fix: Enable prettier's pre-commit hook on scss files

### PR DESCRIPTION
**Pull request by KrishnaPravin**
**Fix pre-commit hook configuration**
Changes:
- Added the file types to be checked by prettier
- ~~Replace `--write` with `--check` for prettier and removed `--fix`~~.
    - ~~Reason: when the auto-fix is enabled, devs might not be aware of the changes made by these tools and the changes will come along with the commit.~~

**Type of PR**

- [x] Bugfix
- [ ] New feature

**Screenshots**
Screenshot 2020-04-12 at 4 07 56 PM

### CODE REVIEW

**Comment by shuklaayush** [comment]
@KrishnaPravin I think @jeremyphilemon deliberately added the '--write' option to make sure that easy fixes like whitespaces, indentations are fixed in place automatically.

**Comment by KrishnaPravin** [comment]
@shuklaayush I totally agree with whitespace and indentation issues. But what about other code changes? I think auto-fix can make changes to the code without the knowledge of the author of the commit. Do you want that to happen?

**Comment by KrishnaPravin** [comment]
Moreover, whitespace, indentation and all other issues can be fixed by running the respective npm scripts. its just `npm run eslint` and `npm run format`

**Comment by shuklaayush** [comment]
> @shuklaayush I totally agree with whitespace and indentation issues. But what about other code changes? I think auto-fix can make changes to the code without the knowledge of the author of the commit. Do you want that to happen?

@KrishnaPravin Let's just wait for @jeremyphilemon before we merge. I'm not an expert on prettier. I assumed that it only does formatting changes. 

> Moreover, whitespace, indentation and all other issues can be fixed by running the respective npm scripts. its just npm run eslint and npm run prettier

This would work. But what I'm afraid is that a lot of people will instead ignore the prettier errors and just make a PR by adding --no-verify.

**Comment by tannakartikey** [comment]
@KrishnaPravin I think we should put back `--write`. The auto-fix will not change code logic or anything. It will just prettify.

**Comment by KrishnaPravin** [comment]
@shuklaayush @tannakartikey reverted back to auto fix

**Comment by tannakartikey** [comment]
@KrishnaPravin started working perfectly for `scss` files.
Would you like to add the lint task in `.github/workflows/ci.yml` for the server-side check?

**Comment by KrishnaPravin** [comment]
@tannakartikey Yes Sure. I will add the check to `ci.yml`

File: package.json
Comment (tannakartikey) [inline package.json]: @KrishnaPravin from the docs of `sass-lint-auto-fix`

> By default, sass-lint will look for a .sass-lint.yml or .sasslintrc, or an entry in your package.json. When applying the resolver to a given rule, configuration is parsed from one of the above files..
> https://www.npmjs.com/package/sass-lint-auto-fix#configuring-which-rules-to-run-against

There is no mention of a file named `.sass-lint-auto-fix.yml` to be considered automatically as config. And I did not the see that file provided with `-c` option to the `sass-lint-auto-fix` command.

File: package.json
Comment (tannakartikey) [inline package.json:65]: Do we need to run `scss-lint` again after running `scll-lint:fix`? Would not it throw error if it could not fix something?

File: package.json
Comment (tannakartikey) [inline package.json]: Also, any reason why `--max-warnings` is set to 328? Is that the number of warnings we are getting at the moment?

**Review by tannakartikey — CHANGES_REQUESTED** [review]

File: package.json
Comment (KrishnaPravin) [reply package.json]: Yes, @jeremyphilemon  is working on reducing the number

File: package.json
Comment (KrishnaPravin) [reply package.json:65]: Exactly that is what we want, errors like class-name-format and so many others cannot be fixed automatically. For those cases, we would want to prevent a commit until a manual fix is done.

File: package.json
Comment (KrishnaPravin) [reply package.json:44]: We can enable/disable the resolvers using `sass-lint-auto-fix`. For eg: `no-css-comments` this resolver is disabled using `sass-lint-auto-fix.yml` but a warning is enabled in `sass-lint`.

Comment (tannakartikey) [reply package.json]: @KrishnaPravin I got it. But the file `sass-lint-auto-fix.yml` is not given to the command. Should not we have to do something like below?
```
"scss-lint:fix": "sass-lint-auto-fix 'src/**/*.scss' -d -h -c .sass-lint-auto-fix.yml"
```

Comment (KrishnaPravin) [reply package.json]: I will add `-c ` option

Comment (KrishnaPravin) [reply package.json]: I've added it

**Review by tannakartikey — APPROVED** [review]

**Comment by tannakartikey** [comment]
Looks good to me except `npm ci` failed again.

**Comment by KrishnaPravin** [comment]
> Looks good to me except `npm ci` failed again.

we can try `npm i` if `npm ci` is not reliable.

**Comment by tannakartikey** [comment]
@KrishnaPravin can you please rebase?

**Comment by KrishnaPravin** [comment]
> @KrishnaPravin can you please rebase?

I guess @jeremyphilemon is working on this

**Review by jeremyphilemon — APPROVED** [review]

**Pull request merged**
